game/resources.h

```diff
@@ -136,7 +135,7 @@ class Resources final {
 
     static std::vector<uint8_t>      getFileData(std::string_view name);
     static bool                      getFileData(std::string_view name, std::vector<uint8_t>& dat);
-    static phoenix::buffer           getFileBuffer(std::string_view name);
+    static std::unique_ptr<zenkit::Read> getFileBuffer(std::string_view name);
```

Comment:
Why `unique_ptr` ? Can we still use move-constructor here, to simplify caller code?

Reply:
No, `zenkit::Read` is now itself virtual instead of being indirectly virtual through an internal `phoenix::buffer_backing` member. This removes one level of indirection and thus improves performance. It also makes it easier to write custom I/O streams.